### PR TITLE
remove all magic number in asym_verify_impl.

### LINKED ISSUE #587: crypto review action: remove all magic number.
<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/longlon1/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/longlon1/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">
</head>

<body link="#0563C1" vlink="#954F72">

spdmlib/crypto/spdm_ring | asym_verify() | asym_verify_impl.rs | 89 | 1 | The buffer length is represented   by 141 == 66 * 2 + 8 + 1 without explanation. The code would be easier to   understand and less error prone if a set of well-defined symbolic constants   were used instead. NOTE: Magic numbers are used elsewhere too.
-- | -- | -- | -- | -- | --

</body>

</html>

https://github.com/jyao1/rust-spdm/blob/81a2bc40a14be7dcb7b90add073e6f64e3eddf9d/spdmlib/src/crypto/spdm_ring/asym_verify_impl.rs#L83

### PR DESCRIPTION
**Pull request by longlongyang**
fix #587

### CODE REVIEW

File: spdmlib/src/crypto/spdm_ring/asym_verify_impl.rs [hunk 1]
```diff
@@ -80,7 +80,9 @@ fn asym_verify(
             match base_asym_algo {
                 SpdmBaseAsymAlgo::TPM_ALG_ECDSA_ECC_NIST_P256
                 | SpdmBaseAsymAlgo::TPM_ALG_ECDSA_ECC_NIST_P384 => {
-                    let mut der_signature = [0u8; 66 * 2 + 8 + 1];
+                    // DER has this format: 0x30 size 0x02 r_size 0x00 [r_size] 0x02 s_size 0x00 [s_size]
+                    let mut der_signature =
+                        [0u8; crate::protocol::ECDSA_ECC_NIST_P384_KEY_SIZE + 8];
```
Comment (jyao1) [inline spdmlib/src/crypto/spdm_ring/asym_verify_impl.rs:85]: why you change "+ 8 + 1" to "+ 8" ?

File: spdmlib_crypto_mbedtls/src/asym_verify_impl.rs [hunk 2]
```diff
@@ -31,7 +31,8 @@ fn asym_verify(
         }
     };
 
-    let mut der_signature = [0u8; 66 * 2 + 8 + 1];
+    // DER has this format: 0x30 size 0x02 r_size 0x00 [r_size] 0x02 s_size 0x00 [s_size]
+    let mut der_signature = [0u8; spdmlib::protocol::ECDSA_ECC_NIST_P384_KEY_SIZE + 8];
```
Comment (jyao1) [inline spdmlib_crypto_mbedtls/src/asym_verify_impl.rs:35]: why you change "+ 8 + 1" to "+ 8" ?

File: spdmlib/src/crypto/spdm_ring/asym_verify_impl.rs (hunk 1 above)
Comment (longlongyang) [reply spdmlib/src/crypto/spdm_ring/asym_verify_impl.rs:85]: > why you change "+ 8 + 1" to "+ 8" ?

what does the "+ 1" mean? the DER format should be something like `0x30 size 0x02 r_size 0x00 [r_size] 0x02 s_size 0x00 [s_size]`, the [r_size] + [s_size] should less than or equal to crate::protocol::ECDSA_ECC_NIST_P384_KEY_SIZE, the 8 is for `0x30 size 0x02 r_size 0x00 0x02 s_size 0x00`. Don't know why need to + 1 more byte.

Comment (jyao1) [reply spdmlib/src/crypto/spdm_ring/asym_verify_impl.rs:85]: if the first byte starts from 0x80, then we need append 0x00 before it.

Comment (longlongyang) [reply spdmlib/src/crypto/spdm_ring/asym_verify_impl.rs:85]: > if the first byte starts from 0x80, then we need append 0x00 before it.

the two 0x00 in this `0x30 size 0x02 r_size 0x00 [r_size] 0x02 s_size 0x00 [s_size]` are optianal, and exist only when the first byte >= 0x80. So this case is already counted in for this purpose. We don't need the + 1.

Comment (jyao1) [reply spdmlib/src/crypto/spdm_ring/asym_verify_impl.rs:85]: OK. It seems +8 is right.
I forget why it is +8+1 before.

**Review by jyao1 — APPROVED** [review]

**Pull request merged**